scheduler/src/cook/cached_queries.clj

```diff
@@ -16,3 +19,20 @@
   "Given a job entity, return the user the job runs as."
   [job-ent]
   (caches/lookup-cache-datomic-entity! caches/job-ent->user-cache :job/user job-ent))
+
+(defn instance-uuid->job-uuid-datomic-query
+  "Queries for the job uuid from an instance uuid.
+   Returns nil if the instance uuid doesn't correspond
+   to a job"
+  [db instance-uuid]
+  (->> (d/entity db [:instance/task-id (str instance-uuid)])
+       :job/_instance
+       :job/uuid))
+
+(let [miss-fn
+      (fn [instance-uuid]
+        (instance-uuid->job-uuid-datomic-query (d/db datomic/conn) instance-uuid))]
+  (defn instance-uuid->job-uuid-cache-lookup
```

Comment:
Still planning on removing the `-cache-lookup` from this one?

Reply:
I decided not to because I want it to be clear that this particular function getting `job-uuid` from `instance-uuid` is looking in a cache as opposed to looking straight t datomic. And for that, I thought it made more sense to have:

`instance-uuid->job-uuid-datomic-query` & `instance-uuid->job-uuid-cache-lookup`

vs.

`instance-uuid->job-uuid-datomic-query` & `instance-uuid->job-uuid`

Imo, the former makes it more clear what each function does. Thoughts?

Reply:
I don't feel too strongly one way or the other. 

I definitely see your point, but it's not consistent with the other two existing functions in `cached-queries` (`job->pool-name` and `job-ent->user`). 

I'm fine with either.